File: product_management_ui/src/service/productService.js

//Through this I can do anything like CRUD;

import axios from "axios";

const API_URL="http://localhost:8080";

class ProductService{

postProduct(product){
    return axios.post(API_URL+"/postProduct",product);
}

getAllProduct(){
    return axios.get(API_URL+"/");
}
getProductById(id){
    return axios.get(API_URL+"/"+id);
}

deleteProduct(id){
return axios.get(API_URL+"/deleteProduct/"+id);
}

editProduct(product){
  return axios.post(API_URL+"/editProduct/"+product.id,product);
}



}

export default new ProductService;   

File: product_management_backend/src/main/java/com/ishwar/product_management_backend/service/ProductService.java
package com.ishwar.product_management_backend.service;

import java.util.List;

import com.ishwar.product_management_backend.model.Product;

public interface ProductService {

	public Product saveProduct(Product product);

	public List<Product> getAllProduct();

	public Product getProductById(Integer id);

	public String deleteProduct(Integer id);

	public Product editProduct(Product product,Integer id);
	
}
